src/Components/SuperAdmins/index.jsx

```diff
@@ -4,33 +4,34 @@ import { getSuperAdmins } from '../../redux/superadmins/thunks';
 import { Link } from 'react-router-dom';
 import styles from './super-admins.module.css';
 import Table from './table';
+import Modal from '../Shared/Modal';
 import Button from '../Shared/Button';
 import Loader from '../Shared/Loader';
 
-function SuperAdmins() {
-  const { data, loading, error } = useSelector((state) => state.superadmins);
-  const [message, setMessage] = useState('');
-  const [deleteModal, setDeleteModal] = useState(false);
-  const [confirmModal, setConfirmModal] = useState(false);
+const SuperAdmins = () => {
+  const { loading, error, success, message } = useSelector((state) => state.superadmins);
+  // const [modalMessage, setModalMessage] = useState('');
+  const [showModal, setShowModal] = useState(false);
   const dispatch = useDispatch();
 
-  useEffect(() => {
-    dispatch(getSuperAdmins());
-  }, [dispatch]);
+  const handleModal = () => {
+    setShowModal(!showModal);
+    setTimeout(() => {
+      setShowModal();
+    }, 4000);
+  };
 
-  const deleteItem = async (id) => {
-    try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/${id}`, {
-        method: 'DELETE'
-      });
-      const data = await response.json();
-      setMessage(data.message);
-      setDeleteModal(true);
-      return data;
-    } catch (error) {
-      console.error(error);
+  useEffect(() => {
+    console.log(message);
+    if (success) {
+      handleModal();
     }
-  };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    getSuperAdmins(dispatch);
+  }, [dispatch]);
```

Comment:
dispatch is a constant (function), it doesn't make sense to be listed as a dependency because it doesn't change state